tests/Unit/Shipping/LocationRatesProcessorTest.php

```diff
@@ -22,69 +21,88 @@ class LocationRatesProcessorTest extends UnitTest {
 	public function test_process_returns_only_most_expensive_flat_rate() {
 		$location       = new Location( 'US', 'CA' );
 		$location_rates = [
-			new LocationRate( $location, new ShippingRateFlat( 50 ) ),
-			new LocationRate( $location, new ShippingRateFlat( 100 ) ),
-			new LocationRate( $location, new ShippingRateFlat( 200 ) ),
+			new LocationRate( $location, new ShippingRate( 50 ) ),
+			new LocationRate( $location, new ShippingRate( 100 ) ),
+			new LocationRate( $location, new ShippingRate( 200 ) ),
 		];
-		$processed = $this->rates_processor->process( $location_rates );
+		$processed      = $this->rates_processor->process( $location_rates );
 		$this->assertEquals( 1, count( $processed ) );
-		/** @var ShippingRateFlat $flat_rate */
-		$flat_rate = $processed[0]->get_shipping_rate();
-		$this->assertInstanceOf( ShippingRateFlat::class, $flat_rate );
-		$this->assertEquals( 200, $flat_rate->get_rate() );
+		$this->assertEquals( 200, $processed[0]->get_shipping_rate()->get_rate() );
```

Comment:
❓

Should not return the cheapest value? 50?

Reply:
No it's intentionally returning the most expensive rate as I mentioned in https://github.com/woocommerce/google-listings-and-ads/pull/1422#discussion_r851109599